### PR TITLE
Added new functionality of middleware server power operations, added test cases.

### PR DESCRIPTION
**Pull request by hhovsepy**
Fixed possible bugs with setting attributes from kwargs['properties'], now it checks the existence of of attribute, as kwargs['properties'] can contain attributes like "name"  and "hostname", which are for MiQ internal use.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 60c8c6c6399efccd7c343a7e3b08a4de19dc6eec:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d3bceb6e61568603bbad680cc770b8d2cea540a8:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by hhovsepy** [comment]
In fact tests are skipped, because of "ERROR:cfme:Could not do is_refreshed at /cfme_tests_te/utils/varmeth.py:74 in time"

**Comment by dajoRH** [comment]
Lint report for commit 0da0320f0f9dd7900f2a8c3f1b46984fa9803839:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c2371dc5ba90bbb08157a47cf7a35748c71f6b20:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 889d109b0b30b4287a1ff9715a171f4a222cbe6d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3145460b7f86bf18e19e4f8be079e4e7e8dd0063:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3f9fb94b75bd4303018308373854e1a379740ad1:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/server.py [hunk 1]
```diff
@@ -100,7 +101,9 @@ def servers(cls, provider=None, strict=True):
                         _provider = get_crud(get_provider_key(row.provider.text))
                     servers.append(MiddlewareServer(name=row.server_name.text,
```
Comment (psav) [inline cfme/middleware/server.py]:
```python
                    servers.append(MiddlewareServer(
                        name=row.server_name.text,
                        feed=row.feed.text,
                        hostname=row.host_name.text,
                        product=row.product.text or None,
                        provider=_provider))
```

**Comment by dajoRH** [comment]
Lint report for commit 8b57e9a782d70cee2abd7206600d6a5f7cb08cbf:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/server.py (hunk 1 above)
Comment (hhovsepy) [inline cfme/middleware/server.py]: ok, thanks for hint, will use this format for all

**Comment by psav** [comment]
Code wise, looks good, but would love to get something passing through PRT

**Comment by dajoRH** [comment]
Lint report for commit 939d324d1a3af4ada92d5dce79ff3f1b6cf695ea:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**